src/report_reader.cpp

```diff
@@ -359,19 +355,21 @@ DataFrame<T> ReportReader<T>::Population::get(const Selection& selection,
             .select({index_start, it->second.first},
                     {index_stop - index_start + 1, it->second.second - it->second.first})
             .read(data);
+        int index = 0;
+        for (auto& datum : data) {
+            for (auto d : datum) {
+                data_frame.data[index].push_back(d);
+            }
```

Comment:
I think changing the names would be beneficial for readability. 
I am a bit lost with the indexing order... 

datum is column or line from the initial 2D array ?
index is ids or times ?

Reply:
You can also remove the second auto --> float. This makes clear why you don `&`